homeassistant/components/decora/light.py

```diff
@@ -58,7 +59,10 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     lights = []
     for address, device_config in config[CONF_DEVICES].items():
         device = {}
-        device["name"] = device_config[CONF_NAME]
+        try:
+            device["name"] = device_config[CONF_NAME]
+        except KeyError:
+            device["name"] = util.slugify(address)
```

Comment:
Add a custom voluptuous config validator instead that sets the name item if it's missing in the device config. See eg:
https://github.com/home-assistant/home-assistant/blob/6ac7796fb7599565cf44e90fad1efefa56c43d5a/homeassistant/components/manual/alarm_control_panel.py#L63-L75

Reply:
Sure thing!

Reply:
Do I have to create a complete new schema or can I just extend the light platform schema and add this custom config validator?

Reply:
I have tried to extend the light platfrom schema, but without any luck so far.
Building a new one works though:
```python
DEVICE_SCHEMA = vol.Schema(
    {vol.Optional(CONF_NAME): cv.string, vol.Required(CONF_API_KEY): cv.string}
)

PLATFORM_SCHEMA = vol.Schema(
    vol.All(
        {
            vol.Required(CONF_PLATFORM): "decora",
            vol.Optional(CONF_DEVICES, default={}): {cv.string: DEVICE_SCHEMA},
        },
        _name_validator,
    )
)
```
@MartinHjelmare am I missing something? :)

Reply:
Something like this:
```py
PLATFORM_SCHEMA = vol.Schema(
    vol.All(
        PLATFORM_SCHEMA.extend({...}),
        _name_validator
    )
)
```

Reply:
Yeah you are completely right, did not think of that. My bad 😁 solved with https://github.com/home-assistant/home-assistant/pull/28279/commits/994ec3e4b47d9b70c3d1c01d85fa4b0b59f271cb